librz/core/cprint.c

```diff
@@ -505,3 +505,369 @@ RZ_API RZ_OWN char *rz_core_print_bytes_with_inst(RZ_NONNULL RzCore *core, RZ_NO
+	rz_config_set_i(core->config, "asm.offset", true);
+	rz_config_set_i(core->config, "asm.dwarf", true);
+	rz_config_set_i(core->config, "scr.color", COLOR_MODE_DISABLED);
+	rz_config_set_i(core->config, "asm.tabs", 0);
+	rz_config_set_i(core->config, "scr.html", 0);
+	rz_config_set_i(core->config, "asm.cmt.right", true);
+
+	char *dump_string = NULL;
+	RzList *lines = NULL;
+	if (!strncmp(input, "dsb", 3)) {
+		RzAnalysisBlock *bb = rz_analysis_find_most_relevant_block_in(core->analysis, core->offset);
+		if (bb) {
+			dump_string = rz_core_cmd_strf(core, "pD %" PFMT64u " @ 0x%08" PFMT64x, bb->size, bb->addr);
+		}
+	} else if (!strncmp(input, "dsf", 3) || !strncmp(input, "dsr", 3)) {
+		RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, core->offset, RZ_ANALYSIS_FCN_TYPE_NULL);
+		if (fcn) {
+			dump_string = rz_core_cmd_str(core, "pdr");
+		} else {
+			eprintf("Cannot find function.\n");
+			goto restore_conf;
+		}
+	} else if (!strncmp(input, "ds ", 3)) {
+		dump_string = rz_core_cmd_strf(core, "pD %s", input + 3);
+	} else {
+		dump_string = rz_core_cmd_str(core, "pd");
+	}
+
+	lines = rz_str_split_duplist(dump_string, "\n", true);
+	if (!lines) {
+		goto restore_conf;
+	}
+	int count = (int)rz_list_length(lines);
+	if (count < 1) {
+		goto restore_conf;
+	}
+	RzListIter *iter;
+	char *line;
+	char *switchcmp = NULL;
+	rz_list_foreach (lines, iter, line) {
+		ut64 addr = UT64_MAX;
+		const char *ox = strstr(line, "0x");
+		if (ox) {
+			addr = rz_num_get(NULL, ox);
+		}
+		const char *qo = strchr(line, '\"');
+		const char *linecolor = NULL;
+		char *string = NULL;
+		if (qo) {
+			char *qoe = strrchr(qo + 1, '"');
+			if (qoe) {
+				int raw_len = qoe - qo - 1;
+				int actual_len = 0;
+				const char *ptr = qo + 1;
+				for (; ptr < qoe; ptr++) {
+					if (*ptr == '\\' && ptr + 1 < qoe) {
+						int body_len;
+						switch (*(ptr + 1)) {
+						case 'x':
+							body_len = 3;
+							break;
+						case 'u':
+							body_len = 5;
+							break;
+						case 'U':
+							body_len = 9;
+							break;
+						default:
+							body_len = 1;
+						}
+						for (int i = 0; i < body_len && ptr < qoe; i++) {
+							ptr++;
+						}
+					}
+					actual_len++;
+				}
+				if (actual_len > 2) {
+					string = rz_str_ndup(qo, raw_len + 2);
+				}
+				linecolor = RZ_CONS_COLOR(comment);
+			}
+		}
+		ox = strstr(line, "; 0x");
+		if (!ox) {
+			ox = strstr(line, "@ 0x");
+		}
+		char *string2 = NULL;
+		if (ox) {
+			char *qoe = strchr(ox + 3, ' ');
+			if (!qoe) {
+				qoe = strchr(ox + 3, '\x1b');
+			}
+			int len = qoe ? qoe - ox : strlen(ox + 3);
+			string2 = rz_str_ndup(ox + 2, len - 1);
+			if (rz_num_get(NULL, string2) < 0x100) {
+				RZ_FREE(string2);
+			}
+		}
+		char *str = NULL;
+		if (asm_flags) {
+			str = strstr(line, ";-- ");
+			if (str) {
+				if (!rz_str_startswith(str + 4, "case")) {
+					rz_strbuf_appendf(sb, "%s\n", str);
+				}
+			}
+		}
+#define USE_PREFIXES 1
+#if USE_PREFIXES
```

Comment:
I know it was already like this, but please remove the unused code and the macro if possible.